### ISSUE TITLE
Update or Add New Email: Notice of Change of Email Address - Party

### ISSUE THREAD

**Issue by JessicaMarine** [ustaxcourt/ef-cms#7674]
As the Court, in order to create a transparent record of when a party's email is first added to the system or an existing address is changed, a Notice of Change of Email Address should be generated by DAWSON.

## Acceptance Criteria
* If a party already has a verified email address on their account and they update that email address to an **unverified** email address:
  * A Notice of Change of Email Address (new code: NOCE) should be generated after the verification process is complete and should be added to the docket record in all open cases and any case closed for 6 months or less
~~* If a party already has a verified email address on their account and they update that email address to another **verified** email address:~~
  ~~* A Notice of Change of Email Address (new code: NOCE) should be generated as soon as the new verified email address is saved and should be added to the docket record in all open cases and any case closed for 6 months or less~~
* If an Admissions Clerk adds a new **unverified** email address to a party's contact information
  * A Notice of Change of Email Address (new code: NOCE) should be generated after the verification process is complete and should be added to the docket record in all open cases and any case closed for 6 months or less
* If a Admissions Clerk adds a new **verified** email address to a party's contact information 
  * A Notice of Change of Email Address (new code: NOCE) should be generated as soon as the new verified email address is saved and should be added to the docket record in all open cases and any case closed for 6 months or less

* For all scenarios above:
  * The Notice of Change of Email Address should have the same logic as a Notice of Change of Address
   * Electronically served in cases with electronic service
   * Work Item created for Docket if paper service or unrepresented petitioner in case
* Notices aren't generated in cases where the status is New

## Security Considerations
 - [ ] Does this work make you nervous about privacy or security?
 - [ ] Does this work make major changes to the system?
 - [ ] Does this work implement new authentication or security controls?
 - [ ] Does this work create new methods of authentication, modify existing security controls, or explicitly implement any security or privacy features?

## Notes
* Chris to look into use case re: whether we generate the notice when they change it in the system OR when they verify their address via email
* Also clarify users (Chris)
* If petitioners AND counsel, create separate stories

## Definition of Done (Updated 4-14-21)
**Product Owner**
 - [ ]  Acceptance criteria have been met and validated on the Court's migration environment

**UX**
 - [x] Business test scenarios to meet all acceptance criteria have been written
 - [x] Usability has been validated
 - [ ] Wiki has been updated (if applicable) 
 - [ ] Story has been tested on a mobile device (for external users only)
 - [ ] Add scenario to testing document, if applicable (https://docs.google.com/spreadsheets/d/1FUHKC_YrT-PosaWD5gRVmsDzI1HS_U-8CyMIb-qX9EA/edit?usp=sharing)

**Engineering**
 - [x] Automated test scripts have been written
 - [x] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [x] Verify that language for docket record for internal users and external users is identical
 - [x] New screens have been added to pa11y scripts
 - [x] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html 
 - [x] READMEs, other appropriate docs, JSDocs and swagger/APIs fully updated
 - [x] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required)
 - [x] Interactors should validate entities before calling persistence methods
 - [x] Code refactored for clarity and to remove any known technical debt
 - [x] Acceptance criteria for the story has been met
 - [ ] Deployed to any Flexion environment for UX Review
 - [ ] Deployed to the Court's migration environment

**Comment by cholly75** [comment]
@mariahkannenberg Updated the pre-condition to answer the questions - notice should be generated when the verification process is complete regardless of the user that initiated it.

**Comment by mariahkannenberg** [comment]
Needs More Info - Need Event Code

**Comment by cholly75** [comment]
@mariahkannenberg - just now edited the story to include new event code NOCE

**Comment by klohman** [comment]
## Notice of Change of Email Docket Entry 

![image.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/95f4718d-ca13-44ce-a6d0-e38b272c90bb)

**Comment by klohman** [comment]
### 7643 Party Changes Email and Notice is Generated 

**Note: Court users do not currently have the ability edit a party’s email address - only the party can make that update.**

#### 1) Unrepresented petitioner updates and does not verify email and a Notice of Email Change is not generated
* Login as petitioner 
    * Petitioner is not represented by counsel 
    * Has cases 
        * In New status
        * In General Docket or other 'open' status 
        * Closed (for less than 6 months) 
* Update email to unverified address (do not verify) 
Expected results 
- [x] Email update shows as pending in UI (Note: Pending email does not show when logged in as public user)
- [x] Notice of Change of Email Address is not generated in any cases 

#### 2) Unrepresented Petitioner updates and verifies email and a Notice of Email Change is generated upon email verification
* Login as petitioner 
    * Petitioner is not represented by counsel 
    * Has cases 
        * In New status
        * In General Docket or other 'open' status 
        * Closed (for less than 6 months) 
* Update to unverified address and verify email 
Expected results 
- [x] New email displays in UI 
- [x] Notice of Change of Email Address is generated in case in General Docket upon email verification
- [x] Notice of Change of Email Address is generated in Closed case upon email verification
- [x] Notice of Change of Email Address is not generated in New case upon email verification
- [x] Notice displays on Docket Record as “Notice of Change of Email Address for [petitioner]”
- [x] Notice is served on all electronic service parties 
- [x] Notice of Change of Email Address work item is created in Docket Document QC Inbox 

#### 3) Unrepresented Intervenor updates and does not verify email and a Notice of Email Change is not generated 
* Login as intervenor 
    * intervenor is not represented by counsel 
    * Has cases 
        * In New status
        * In General Docket or other 'open' status 
        * Closed (for over 6 months) 
* Update to unverified address email (do not verify) 
Expected results 
- [x] Email update shows as pending in UI 
- [x] Notice of Change of Email Address is not generated in any cases 

#### 4) Unrepresented Intervenor updates and verifies email and a Notice of Email Change is generated upon email verification
* Login as intervenor 
    * intervenor is not represented by counsel 
    * Has cases 
        * In New status
        * In General Docket or other 'open' status 
        * Closed (for over 6 months) 
* Update to unverified address and verify email 
Expected results 
- [x] New email displays in UI 
- [x] Notice of Change of Email Address is generated in case in General Docket upon email verification
- [x] Notice of Change of Email Address is not generated in Closed case upon email verification
- [x] Notice of Change of Email Address is not generated in New case upon email verification
- [x] Notice displays on Docket Record as “Notice of Change of Email Address for [intervenor]”
- [x] Notice is served on all electronic service parties 
- [x] Notice of Change of Email Address work item is created in Docket Document QC Inbox 

#### 5) Unrepresented Participant updates and does not verify email and a Notice of Email Change is not generated
* Login as participant 
    * Participant is not represented by counsel 
    * Has cases 
        * In New status
        * In General Docket or other 'open' status 
        * Closed (for less than 6 months) 
* Update to unverified address email (do not verify) 
Expected results 
- [x] Email update shows as pending in UI 
- [x] Notice of Change of Email Address is not generated in any cases 

#### 6) Unrepresented Participant updates and verifies email and a Notice of Email Change is generated upon email verification
* Login as participant 
    * is not represented by counsel 
    * Has cases 
        * In New status
        * In General Docket or other 'open' status 
        * Closed (for less than 6 months) 
* Update to unverified address and verify email 
Expected results 
- [x] New email displays in UI 
- [x] Notice of Change of Email Address is generated in case in General Docket upon email verification
- [x] Notice of Change of Email Address is generated in Closed case upon email verification
- [x] Notice of Change of Email Address is not generated in New case upon email verification
- [x] Notice displays on Docket Record as “Notice of Change of Email Address for [participant]”
- [x] Notice is served on all electronic service parties 
- [x] Notice of Change of Email Address work item is created in Docket Document QC Inbox 

#### 7) Represented Petitioner updates and verifies email and a Notice of Email Change is generated but no Work Item is created upon email verification
* Login as petitioner 
    * is represented by counsel 
    * Has cases 
        * In New status
        * In General Docket or other 'open' status 
        * Closed (for less than 6 months) 
* Update to unverified address and verify email 
Expected results 
- [x] New email displays in UI 
- [x] Notice of Change of Email Address is generated in case in General Docket upon email verification
- [x] Notice of Change of Email Address is generated in Closed case upon email verification
- [x] Notice of Change of Email Address is not generated in New case upon email verification
- [x] Notice displays on Docket Record as “Notice of Change of Email Address for [petitioner]”
- [x] Notice is served on all electronic service parties 
- [x] Notice of Change of Email Address work item is not created in Docket Document QC Inbox 
Note: Work items are created for notices in cases that have any paper service party or any unrepresented party. I do not see a work item being created for a case with only represented, electronic service parties. @cholly75 I believe this is the intended behavior. 

#### 8) Represented intervenor updates and verifies email and a Notice of Email Change is generated but no work item is created upon email verification
* Login as intervenor 
    * is represented by counsel 
    * Has cases 
        * In New status
        * In General Docket or other 'open' status 
        * Closed (for less than 6 months) 
* Update to unverified address and verify email 
Expected results 
- [x] New email displays in UI 
- [x] Notice of Change of Email Address is generated in case in General Docket upon email verification
- [x] Notice of Change of Email Address is generated in Closed case upon email verification
- [x] Notice of Change of Email Address is not generated in New case upon email verification
- [x] Notice displays on Docket Record as “Notice of Change of Email Address for [intervenor]”
- [x] Notice is served on all electronic service parties 
- [x] Notice of Change of Email Address does not create work item for Docket Document QC Inbox 
Note: Work items are created for notices in cases that have any paper service party or any unrepresented party. I do not see a work item being created for a case with only represented, electronic service parties. I believe this is the intended behavior. 

#### 9) Represented participant updates and verifies email and a Notice of Email Change is generated but no Work Item is created upon email verification
* Login as participant 
    * is represented by counsel 
    * Has cases 
        * In New status
        * In General Docket or other 'open' status 
        * Closed (for less than 6 months) 
* Update to unverified address and verify email 
Expected results 
- [x] New email displays in UI 
- [x] Notice of Change of Email Address is generated in case in General Docket upon email verification
- [x] Notice of Change of Email Address is generated in Closed case upon email verification
- [x] Notice of Change of Email Address is not generated in New case upon email verification
- [x] Notice displays on Docket Record as “Notice of Change of Email Address for [participant]”
- [x] Notice is served on all electronic service parties 
- [x] Notice of Change of Email Address does not create work item for Docket Document QC Inbox 
Note: Work items are created for notices in cases that have any paper service party or any unrepresented party. I do not see a work item being created for a case with only represented, electronic service parties. I believe this is the intended behavior. 

#### 10) Unrepresented petitioner updates and does not verify email and a Notice of Email Change is not generated immediately 
* Login as petitioner 
    * Petitioner is not represented by counsel 
    * Has cases 
        * In New status
        * In General Docket or other 'open' status 
        * Closed (for less than 6 months) 
* Update email to verified address (do not verify) 
Expected results 
- [x] Email update shows as pending in UI 
- [x] Notice of Change of Email Address is not generated in any cases 

#### 11) Unrepresented Petitioner updates to email to verified address and a Notice of Email Change is generated immediately
* Login as petitioner 
    * Petitioner is not represented by counsel 
    * Has cases 
      * In New status
      * In General Docket or other 'open' status 
      * Closed (for less than 6 months) 
* Update to verified address 
Expected results 
- [x] New email displays in UI 
- [x] Notice of Change of Email Address is generated in case in General Docket immediately
- [x] Notice of Change of Email Address is generated in Closed case immediately
- [x] Notice of Change of Email Address is not generated in New case immediately
- [x] Notice displays on Docket Record as “Notice of Change of Email Address for [petitioner]”
- [x] Notice is served on all electronic service parties 
- [x] Notice of Change of Email Address work item is created in Docket Document QC Inbox 

#### 12) Admissions Clerk updates unrepresented party's email to unverified address (and address is not later verified): a Notice of Email Change is not generated immediately 
* Login as Admissions Clerk 
  * Update email of unrepresented petitioner with cases 
     that are 
      * In New status
      * In General Docket or other 'open' status 
      * Closed (for less than 6 months) 
* Update email to unverified address (do not verify) 
Expected results 
- [x] Email update shows as pending in UI 
- [x] Notice of Change of Email Address is not generated in any cases 

#### 13) Admissions Clerk updates unrepresented party's email to unverified address (and address is later verified): a Notice of Email Change is generated upon verification 
* Login as Admissions Clerk 
  * Update email of unrepresented petitioner with cases 
     that are 
      * In New status
      * In General Docket or other 'open' status 
      * Closed (for less than 6 months) 
* Update email to unverified address (party logs in a verifies) 
Expected results 
- [x] New email displays in UI 
- [x] Notice of Change of Email Address is generated in case in General Docket upon email verification
- [x] Notice of Change of Email Address is generated in Closed case immediately
- [x] Notice of Change of Email Address is not generated in New case upon email verification
- [x] Notice displays on Docket Record as “Notice of Change of Email Address for [petitioner]”
- [x] Notice is served on all electronic service parties 
- [x] Notice of Change of Email Address work item is created in Docket Document QC Inbox 

#### 14) Admissions Clerk updates unrepresented party's email to verified address: a Notice of Email Change is generated immediately 
* Login as Admissions Clerk 
  * Update email of unrepresented petitioner with cases 
     that are 
      * In New status
      * In General Docket or other 'open' status 
      * Closed (for less than 6 months) 
* Update email to verified address 
Expected results 
- [x] New email displays in UI 
- [x] Notice of Change of Email Address is generated in case in General Docket immediately
- [x] Notice of Change of Email Address is generated in Closed case immediately
- [x] Notice of Change of Email Address is not generated in New case immediately
- [x] Notice displays on Docket Record as “Notice of Change of Email Address for [petitioner]”
- [x] Notice is served on all electronic service parties 

#### 15) Admissions Clerk updates represented party's email to unverified address: a Notice of Email Change is generated upon email verification
* Login as Admissions Clerk 
  * Update email of represented petitioner with cases 
     that are 
      * In New status
      * In General Docket or other 'open' status 
      * Closed (for less than 6 months) 
* Update to unverified address (party logs in a verifies) 
Expected results 
- [x] New email displays in UI 
- [x] Notice of Change of Email Address is generated in case in General Docket upon email verification
- [x] Notice of Change of Email Address is generated in Closed case upon email verification
- [x] Notice of Change of Email Address is not generated in New case upon email verification
- [x] Notice displays on Docket Record as “Notice of Change of Email Address for [petitioner]”
- [x] Notice is served on all electronic service parties 
- [x] Notice of Change of Email Address work item is not created in Docket Document QC Inbox 

#### 16) Admissions Clerk updates represented party's email to unverified address: a Notice of Email is not generated
* Login as Admissions Clerk 
  * Update email of represented petitioner with cases 
     that are 
      * In New status
      * In General Docket or other 'open' status 
      * Closed (for less than 6 months) 
* Update to unverified address (do not verify) 
Expected results 
- [x] Email update shows as pending in UI 
- [x] Notice of Change of Email Address is not generated in any cases 

#### 17) Admissions Clerk updates represented party's email to verified address: a Notice of Email Change is generated immediately 
* Login as Admissions Clerk 
  * Update email of represented petitioner with cases 
     that are 
      * In New status
      * In General Docket or other 'open' status 
      * Closed (for less than 6 months) 
* Update to verified address
Expected results 
- [x] New email displays in UI 
- [x] Notice of Change of Email Address is generated in case in General Docket immediately
- [x] Notice of Change of Email Address is generated in Closed case immediately
- [x] Notice of Change of Email Address is not generated in New case immediately
- [x] Notice displays on Docket Record as “Notice of Change of Email Address for [petitioner]”
- [x] Notice is served on all electronic service parties 
- [x] Notice of Change of Email Address work item is not created in Docket Document QC Inbox - work items are created in cases with paper service or unrepresented petitioners

**Comment by kkoskelin** [comment]
Business Logic Questions:
 * if petitioner on case did not previously have e-access (no previous email), should NOCE be filed?
   * does "update" (in story summary) mean only when updating from one defined value to another, and _not_ when setting an initial value from empty?
 * design guidance requested for formatting of NOCE pdf (see screenshot below)
 
![Screen Shot 2021-09-13 at 1 41 14 PM](https://user-images.githubusercontent.com/2445917/133146925-614c1ffa-fb2f-4625-9aac-1c323088dda4.png)

**Comment by cholly75** [comment]
@klohman @kkoskelin What do we do when adding a phone number to a (not NEW) case where none previously existed (or N/A existed)? I should think we would want to follow the same logic here.

**Comment by klohman** [comment]
@cholly75 The phone number field is required, so it really always exists (it just may say "No telephone number" or "N/A'. If we want to follow the same logic, that means every time a petitioner is granted eAccess to a new case, this Notice will be generated, if that's the desired behavior. The Notice itself than could just have "No email provided" as the previous email. 
@kkoskelin

**Comment by cholly75** [comment]
@klohman I think that's OK, since this will only affect cases not in NEW status, then changing from an absence of an email to the presence of an email would warrant a notice. I like the "no email provided" verbiage for the prior email address in the notice.

**Comment by kkoskelin** [comment]
@cholly75 This story seems to have been focused primarily on the user-verification process which would generate the NOCE.  However, there is another scenario in which the existing user in cognito is assigned to a case.  For this scenario, no verification is required.  However, this scenario is also not covered in the acceptance criteria.

Should this new scenario (in which an existing eAccess account -- and email address -- is in the system and no longer requires verification) be handled in a new story, or do we want to add AC to this ongoing story here?

**Comment by TomElliottFlexion** [comment]
Updated NOCE @klohman ![Screen Shot 2021-09-14 at 11.10.59 AM.png](https://images.zenhubusercontent.com/601ac936c7e347e64fed075e/a97ea8aa-a230-44a6-a2e6-c4b78e356a4e)

**Comment by cholly75** [comment]
@kkoskelin That would be a separate story; this is concerned with the change of a party's email address, adding a party to a case would be a different scenario, unless I'm missing some nuance here?

**Comment by kkoskelin** [comment]
@cholly75 I think we'll need to discuss further -- as mentioned in a previous message, this workflow has been focused on the verification of email as the trigger (which is sometimes but not always the product of "change of a party's email address"), without regard to what process triggered it.  Our work on this story thus far makes no distinction between the scenarios you mention.

**Comment by mariahkannenberg** [comment]
BLOCKED - There are some nuances we need to discuss with @cholly75 around the AC referenced above before we unblock.

**Comment by JessicaMarine** [comment]
@cholly75 I just worked with the team to re-work the AC.  Because the generation of the Notice of Change of Email Address was tied to the verification process, there is no real easy way to bifurcate the verification that occurs when a party changes their email address versus the verification that is required when Admissions adds a new email address for a party.  As a result, some additional work needed to be brought back into this story.  We basically ended up with 4 scenarios:

- Party updates email address to an unverified email;
- Party updates email address to another verified email;
- Admissions Clerk adds an unverified email to a party; and
- Admissions Clerk adds a verified email to a party.

**Comment by TomElliottFlexion** [comment]
Upon further investigation, it seems that it is not possible for a party to update their own email address to one already in use (AC bullet 2) ![Screen Shot 2021-09-14 at 4.33.40 PM.png](https://images.zenhubusercontent.com/601ac936c7e347e64fed075e/4748c6d0-aa50-4598-91dc-c60961fb1465) @JessicaMarine @cholly75

**Comment by mariahkannenberg** [comment]
Still blocked - Technical incompatibility with AC ustaxcourt/ef-cms#6492. See above last two comments.

**Comment by JessicaMarine** [comment]
@TomElliottFlexion  I took that AC 2 out. It was only included for completeness.

**Comment by TomElliottFlexion** [comment]
~@JessicaMarine @cholly75 More questions! If a verified email has already been added to a party on a case, should that address be blocked from being added to another party? Should we allow multiple parties to be associated with the same verified address (and cognito account)? @kkoskelin @codyseibert~
Note/Open question for future stories: Adding the _same_ **_verified_ **email address to a second (or third, etc) party on a case currently associates the two with the same cognito account, and therefore the same userId. This creates a few problems, e.g. being unable to edit the second petitioner's contact information. We are working around this to maintain current functionality in this story, but we may want to examine this again in a future story.

**Comment by mariahkannenberg** [comment]
Tom, Kris, Cody and Joshua working this today. 
Jessica mentioned that this notice doesn't have a title, like the rest of them do. Add the title to the design. Tom and folks to add. Chris and Tom to also talk about question of service and backlog.

**Comment by klohman** [comment]
@ttlenard @TomElliottFlexion  Notice of Change of Address does not have a title currently (on migration) so that's probably where the template came from for the new notice. We'll need to update both (I assume) to include the title above the "Please change the contact information..." line

![Screen Shot 2021-09-21 at 10.43.02 AM.png](https://images.zenhubusercontent.com/5bd7271508843955cc24d3d0/d84107f6-dc07-440a-85b8-b6605e473a7b)

**Comment by TomElliottFlexion** [comment]
@klohman @ttlenard  We currently add the title to the coversheet for all `Notice of Change` documents, should we add it to the document itself for all four? ![Screen Shot 2021-09-21 at 9.54.10 AM.png](https://images.zenhubusercontent.com/601ac936c7e347e64fed075e/626cdc68-3313-4328-87ad-66d3d3ef3b19)

**Comment by klohman** [comment]
The title should be on the cover sheet AND the actual Notice. It was there previously, and I'm not sure when it was dropped but I don't think it was intentional.

**Comment by TomElliottFlexion** [comment]
@ttlenard @klohman Updating the template to look like this: (Same as `Notice of Change` docs before ![Screen Shot 2021-09-21 at 11.13.12 AM.png](https://images.zenhubusercontent.com/601ac936c7e347e64fed075e/4f7303a9-f79e-4ed0-a957-6c9732f4ab9f)

**Comment by ttlenard** [comment]
@klohman @TomElliottFlexion Thank for you getting the title added to the template!  It looks great! I'm a little confused by Kristen's first comment in that the title isn't on the Notice of Change of Address in MIG. Last week when I grabbed screen shots of each notice, I'm pretty sure I was in MIG and took this:  

![image.png](https://images.zenhubusercontent.com/60f99b8ac0d8d04c8aed1aa3/d9dfce61-ccb8-41f2-97a3-e786385a10d1)

Are you indicating that the title isn't on that Notice anymore?

**Comment by klohman** [comment]
> @klohman @TomElliottFlexion Thank for you getting the title added to the template! It looks great! I'm a little confused by Kristen's first comment in that the title isn't on the Notice of Change of Address in MIG. Last week when I grabbed screen shots of each notice, I'm pretty sure I was in MIG and took this:
> 
> ![image.png](https://camo.githubusercontent.com/0186797778486dcc02aee090c101eace619d3d360201af7877958a587e46a9af/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3630663939623861633064386430346338616564316161332f64396466636536312d636362382d343166322d393761332d653738363338356131306431)
> 
> Are you indicating that the title isn't on that Notice anymore?

@ttlenard Yes, this morning I checked the other Notices on mig because the Notice of Change of Email would have used the same general template. The title was missing from the other notices as well (likely a recent update made mistakenly), but I belivee @TomElliottFlexion is updating them all to ensure they have the title.

**Comment by TomElliottFlexion** [comment]
@ttlenard I can explain this; Both the `Notice of Change of Address` and the `Notice of Change of Email Address` are generated using the same template- a template we were/are still working on. The changes to `Notice of Change of Address` on migration are a part of this story, and the titles should be back to normal when this story is complete.

**Comment by TomElliottFlexion** [comment]
Thanks for pointing this out, we might not have noticed right away!

**Comment by ttlenard** [comment]
Thanks for the additional information!

**Comment by mariahkannenberg** [comment]
Need someone w/deeper AWS experience. Mariah to track someone down.

**Comment by mariahkannenberg** [comment]
Created during standup as we were chatting. Chris to finesse test cases and update team. 

"For any case that's not new, a party should be able to file a document." 

"In a case that's been closed for greater than 6 months, the user should be able to change their email, but not autogenerate notices."

**Comment by cholly75** [comment]
Regression Tests to be executed:

1) Ensure that cases closed for over 6 months will still allow parties to file and serve documents
2) Ensure that cases closed for over 6 months do not generate Change of Address/Phone/Both notices
3) Add "Cases that are Closed for over 6 months" to party criteria already populated for existing test cases

**Comment by klohman** [comment]
> Regression Tests to be executed:
> 
> 1. Ensure that cases closed for over 6 months will still allow parties to file and serve documents
> 2. Ensure that cases closed for over 6 months do not generate Change of Address/Phone/Both notices
> 3. Add "Cases that are Closed for over 6 months" to party criteria already populated for existing test cases

Reviewed 
> 1. Ensure that cases closed for over 6 months will still allow parties to file and serve documents 
- [x] Worked as expected 
> 2. Ensure that cases closed for over 6 months do not generate Change of Address/Phone/Both notices
- [ ] Does not work as expected. A Notice of Change of Address is being generated in a case closed for more than 6 months. 
> 3. Add "Cases that are Closed for over 6 months" to party criteria already populated for existing test cases
- not sure what this means - @cholly75 can you clarify?

**Comment by cholly75** [comment]
@klohman I was suggesting that in the test cases already here that specify (for example):
"Petitioner Has Cases:
In New status
In General Docket or other 'open' status
Closed (for less than 6 months)"

That we add "Closed (for 6 months or more)" to check behavior in those cases as well when we do review.

**Comment by codyseibert** [comment]
We decided to go ahead and address the > 6 month rule for the following scenarios even though these are new features not regressions:
- a petitioner updating their own address,phone,phone+address
- an internal user updating the address,phone,phone+address on a petitioner

These two flows did NOT have logic related to the > 6 month rule to prevent a notice from generating, but we went ahead and added functionality to prevent the generation since this code is fresh in our heads.  We debated making a story to add these later, but I personally didn't want to have to dive down these code paths at a later point.

**Comment by codyseibert** [comment]
@mmarcotte heads up, for the batch 9 deploy, you'll need to run an environment specific deploy, and also the deploy will probably fail on the web-api terraform run the first time you try to deploy. There is a script called setup-s3-triggers-file.sh which you will need to run to copy the triggers.zip inside the deploys bucket. This is a work around needed due to how the blue green is setup in terraform (it would be nice to figure out a better solution than to run the deploy twice in a row, I can make a opex card for it if needed. It seems low priority since this is a one time setup on a new environment)

**Comment by ttlenard** [comment]
Some initial testing feedback:

1) Test case 4 - to get this test case set up, I needed to log in as a docket clerk, and I had to add in an intervenor to this case: 20078-21.  I then logged in as an admissions clerk to add in an email address.   I did the same thing for 20079-21. This step generates a NOCE that says "no email" to "email". There were 2 duplicate NOCE's were generated on the docket record for 20078-21.  They are docket entry 5 and 6. The other case was fine - only one NOCE.  

2) I am also seeing some strange behavior with service if there is the same intervenor's email on multiple cases. If an intervenor is associated with a case, and then get's added to a different case at a later time, a NOCE is generated in ALL cases they are associated with, even when their email isn't changed.  This one is kinda tricky to explain, and I'd be happy to do a zoom call. 

3) In Test Case 5, I had to get a participant added into the case. I logged in as a docket clerk, added in a participant in cases 20078-21 and 20079-21.  Logged in as admissions clerk and added in an email. It autogenerated notices prior to verification of the email.

4) Test Case 6 - When admissions clerk added in an email for a participant, the docket has 2 NOCE generated on Docket 20078-21 (entry 10, 11). i also got 2 NOCE generated on docket 20079-21 (entry 9, 10). Even though there were two NOCE's generated, i only received one email per case.

**Comment by ttlenard** [comment]
Some testing feedback from 10-12-21. I posted this on Slack, but thought I better add it here as well.

- Test Case 7 - Docket is now getting a work item created when a petitioner is represented by counsel.  This was not happening previously.  This is most likely the going to be the same scenario for Intervenors and Participants that are represented in Test case 8 and 9.

- I also found a small formatting issue on the title of the Notice on the Docket Record.  This is only occurring when an IRS practitioner is filing the first document in a case, and when a Private Practitioner is first filing an Entry of Appearance to get access to the case.  The formatting of the title of the notice is fine everywhere else.

here is a screen grab of when a private practitioner is requesting access to case: ![image.png]
(https://images.zenhubusercontent.com/60f99b8ac0d8d04c8aed1aa3/5ae02e9b-a857-438a-adad-7e64816a7ccf)

here is a screen grab of when an IRS practitioner is filing the first document in a case:
![image.png](https://images.zenhubusercontent.com/60f99b8ac0d8d04c8aed1aa3/0afc697e-861e-4230-8133-71358cdbd873)
